spring-boot-samples/spring-boot-sample-cache/src/main/java/sample/cache/config/GeodeConfiguration.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2011-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sample.cache.config;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.ImportResource;
+
+/**
+ * Spring {@link Configuration @Configuration} class to configure GemFire as a caching provider.
+ * This {@link Configuration @Configuration} class is conditionally loaded based on
+ * whether GemFire is on the classpath.
+ *
+ * @author John Blum
+ * @see org.springframework.boot.autoconfigure.condition.ConditionalOnClass
+ * @see org.springframework.context.annotation.Configuration
+ * @see org.springframework.context.annotation.ImportResource
+ * @since 1.5.0
+ */
+@Configuration
+@ConditionalOnClass(name = { "org.apache.geode.cache.Cache" })
```

Comment:
Perhaps a comprise; I can use a `@Profile` instead?

I think it would be nicer for _Spring Boot_ users if we kept the **commenting** and **uncommenting** to a minimum.  Plus, there is not mention of comment and uncommenting caching sample POM file dependencies in the [README](https://github.com/spring-projects/spring-boot/tree/v1.4.1.RELEASE/spring-boot-samples/spring-boot-sample-cache).

I find it tedious to have to modify the caching sample POM file in order to comment out the current caching provider dependency and uncomment another dependency every time I want to try a different caching solution.

I have changed the sample cache POM file to use Maven profiles.

I have also changed the Geode cache sample configuration to use a _Spring_ profile.